pkg/virt-api/rest/cpu.go

```diff
@@ -0,0 +1,68 @@
+package rest
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+
+	"github.com/emicklei/go-restful/v3"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/kubevirt/pkg/virt-api/definitions"
+)
+
+func (app *SubresourceAPIApp) CPURequestHandler(request *restful.Request, response *restful.Response) {
+	namespace := request.PathParameter(definitions.NamespaceParamName)
+	name := request.PathParameter(definitions.NameParamName)
+	cpuCountStr := request.QueryParameter(definitions.CpuCountParamName)
+
+	if cpuCountStr == "" {
+		writeError(errors.NewBadRequest("cpuCount parameter is required"), response)
+		return
+	}
+
+	cpuCount, err := strconv.Atoi(cpuCountStr)
+	if err != nil {
+		writeError(errors.NewBadRequest("cpuCount parameter must be an integer"), response)
+		return
+	}
+
+	if cpuCount < 0 {
+		writeError(errors.NewBadRequest("cpuCount parameter must be non-negative"), response)
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancel()
+
+	// Validate that the VM exists
+	if _, err = app.virtCli.VirtualMachine(namespace).Get(ctx, name, metav1.GetOptions{}); err != nil {
+		if errors.IsNotFound(err) {
+			writeError(errors.NewNotFound(v1.Resource("virtualmachine"), name), response)
+		} else {
+			writeError(errors.NewInternalError(err), response)
+		}
+		return
+	}
+
+	patchPayload, err := patch.New(
+		patch.WithReplace("/spec/template/spec/domain/cpu/cores", cpuCount),
```

Comment:
https://kubevirt.io/user-guide/compute/cpu_hotplug/#cpu-hotplug

> With CPU hotplug, it is possible to patch the sockets integer in the VM template and the change will take effect right away.

Reply:
After a discussion, it's been decided to implement this feature by using the virtctl client to patch the VM resource directly, without creating new REST API endpoints.